content/08-mongodb/09-mongodb-indexes.mdx

```diff
@@ -0,0 +1,260 @@
+---
+title: 'Introduction to MongoDB indexes'
+metaTitle: "MongoDB Indexes | Prisma's Data Guide"
+metaDescription: "In this guide, we'll cover indexes in MongoDB."
+metaImage: '/content/mongodb/header.png'
+---
+
+## Introduction
+Much like an encyclopedia, a database is a bountiful store of accessible information. An encyclopedia can have the answers to many questions one might have about history, anthropology, geography and so on. 
+
+To answer a specific question or find a specific piece of information in an encyclopedia would require sifting through every page until you come across what you are looking for. Unless you are a speed reader, this is an incredibly inefficient task. That is why an encyclopedia has an index which points you to the exact page you need to turn to for the information you are looking for. 
+
+A database index similarly points you in the right place for the seeked information more efficiently. In MongoDB, a query not using an index that "searches through the whole book" is called a *collection scan*.
+
+Indexes can be thought of as shortcuts for your queries so that the entire database does not need to be scanned to find what you are looking for. In this article, we are going to introduce indexes in MongoDB, when to use them, and how to manage them.
+
+<PrismaOutlinks>
+
+If you are using [Prisma Client](https://www.prisma.io/docs/concepts/components/prisma-client) with MongoDB, you can use the [MongoDB connector (currently in preview)](https://www.prisma.io/docs/concepts/database-connectors/mongodb) to connect and manage your data.
+
+</PrismaOutlinks>
+
+
+## When should you use an index
+Keeping with the our encyclopedia analogy, one might thing to have an index line for every word in the book. If it is always faster to use an index, then this seems beneficial. However, as you can imagine, the more lines of words in an index then the larger the book becomes. At some point the size of the book to accomodate indexing every word becomes ineffective. There are many words such as "the" or "because" that are less useful to search for than "hippopotamus".
+
+This is similar to an index in MongoDB and databases in general. While yes, it is faster to have an index for any possible query, there simply are queries that don't need indexing. Like the size of a book, adding too many indexes to a database also takes up space and has a detrimental effect on write operations on the database if not kept in check.
+
+Indexes are an incredibly useful way to optimize querying your database and are recommended. Knowing when to use them is important, so making sure that you are adding them on database fields that are often being queried will keep your reads efficient without negatively impacting database size and write efficiency.
+
```

Comment:
It might be useful to add a quick sentence explaining that you want to index fields that are explicitly used as selection criteria when retrieving records.  For instance, if you have an ID column or you often search for records by email address, you'd want to add indexes to those fields.